cirq/sim/sparse_simulator.py

```diff
@@ -331,3 +336,27 @@ def sample(self,
                                                     self, None),
                                                 repetitions=repetitions,
                                                 seed=seed)
+
+
+def _split_into_unitary_then_general(circuit: 'cirq.Circuit'
```

Comment:
There is probably a case to be made for a good way to handle this sort of splitting of circuits by disjoint filter as part of Cirq.  No need to do now but maybe we should have an issue for this?

Reply:
Acknowledged.